pvlib/tests/test_tools.py

```diff
@@ -144,3 +147,192 @@ def test_get_pandas_index(args, args_idx):
 def test_normalize_max2one(data_in, expected):
     result = tools.normalize_max2one(data_in)
     assert_allclose(result, expected)
+
+
+@pytest.mark.parametrize(
+    'input,expected',
+    [
+        (
+            {
```

Comment:
The `dict` here appears only to be so that the function call can be double-splatted `(**input)`. I'm interested in feedback from others: is this parameterization style easier or more difficult, as a reviewer? It's not the style used for other tests are parameterized.

Reply:
My 2 cents: a more compact representation of the changing values is helpful for reviewing, and is worth a bit of processing inside the test function itself.  For example, it seems like the following:
```python
(
    {
        "time": pd.DatetimeIndex(
            ["1974-06-22T18:30:15"],
            tz=ZoneInfo("Etc/GMT+5"),
        ),
        "location": location.Location(
            43.19262774396091, -77.58782907414867, tz="Etc/GMT+5"
        )
    },
    pd.DatetimeIndex(["1974-06-22T23:30:15"], tz=ZoneInfo("UTC")),
),
(
    {
        "time": pd.DatetimeIndex(["1974-06-22T18:30:15"]),
        "location": location.Location(
            43.19262774396091, -77.58782907414867, tz="Etc/GMT+5"
        )
    },
    pd.DatetimeIndex(["1974-06-22T23:30:15"], tz=ZoneInfo("UTC")),
),
```

could be replaced with something like this: 
```python        
("1974-06-22T18:30:15", "Etc/GMT+5", 43, -77, "Etc/GMT+5", "1974-06-22T23:30:15+00:00"),
("1974-06-22T18:30:15", None, 43, -77, "Etc/GMT+5", "1974-06-22T23:30:15+00:00"),
```
And the test function could then contain the necessary calls to `Location` and `pd.DatetimeIndex` to construct the parameters themselves.

I think the latter format makes it easier to see what changes between each set of parameters.  I found my eyes having to flick back and forth and subsequently getting lost with the former format.